segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperServiceRunner.java

```diff
@@ -294,24 +326,37 @@ private BookieServerAndResources runBookie(int bkPort) throws Exception {
         }
 
         log.info("Starting Bookie at port " + bkPort);
-        BookieResources resources = new ResourceBuilder(conf).build();
-        Bookie bookie = createBookie(resources);
-        val bs = new BookieServer(conf, bookie, org.apache.bookkeeper.stats.NullStatsLogger.INSTANCE, UnpooledByteBufAllocator.DEFAULT,
-                new UncleanShutdownDetectionImpl(resources.getLedgerDirsManager()));
-        bs.start();
-        return new BookieServerAndResources(bs, resources);
+        BookieResources resources = null;
+        resources = new ResourceBuilder(conf).build();
+        Bookie bookie = null;
+        bookie = createBookie(resources);
+        BookieServer bs = null;
+        try {
+            bs = new BookieServer(conf, bookie, NullStatsLogger.INSTANCE, UnpooledByteBufAllocator.DEFAULT,
+                    new UncleanShutdownDetectionImpl(resources.getLedgerDirsManager()));
+        } catch (SecurityException | InterruptedException | BookieException | KeeperException |
```

Comment:
Would it be easier to check first for the exception you are looking for to handle differently (associated to port binding issue) and if the exception is different, then just throw the `RuntimeException`?

Reply:
done.